src/errum_cpp_routine.cpp

```diff
@@ -0,0 +1,564 @@
+    unsigned int chain_m_burn = chain_length-burnin;
+    unsigned int tmburn;
+    arma::vec vv = bijectionvector(K);
+    arma::mat Observed_ORs = OddsRatio(N,J,Y);
+
+    arma::mat XpX = X.t() * X;
+    arma::vec diagXpX = XpX.diag();
+    unsigned int V = XpX.n_cols;
+
+    arma::mat Q = arma::randi<arma::mat>(J,K,arma::distr_param(0,1));//random_Q(J,K);
+    arma::mat CLtable = CL_invbijection_table(K,nClass);
+
+    arma::mat Z = arma::randn<arma::mat>(J,K);
+    arma::mat Gamma = arma::randn<arma::mat>(V,K);
+    arma::mat deltas = arma::zeros<arma::mat>(V,K);//arma::randi<arma::vec>(V,arma::distr_param(0,K));
+    arma::vec oneK=arma::ones<arma::vec>(K);
+    double w2(1.),nu(1.);
+
+    //Saving output
+    arma::vec deviance(chain_m_burn);
+    arma::mat PISTAR(J,chain_m_burn);
+    arma::cube RSTAR(J,K,chain_m_burn);
+    arma::mat PIs(nClass,chain_m_burn);
+    //arma::vec cv1s(chain_m_burn);
+    //arma::vec v1s(chain_m_burn);
+    //arma::vec v0s(chain_m_burn);
+    arma::cube QS(J,K,chain_m_burn);
+    arma::cube GAMMAS_OUT(V,K,chain_m_burn);
+    arma::mat Delta_tab=arma::zeros<arma::mat>(V,K);
+    arma::cube B0s(J,K,chain_m_burn);
+    arma::cube B1s(J,K,chain_m_burn);
+    arma::mat OR_PPPs=arma::zeros<arma::mat>(J,J);
+
+    //need to initialize parameters
+    arma::vec CLASS = arma::randi<arma::vec>(N,arma::distr_param(0,nClass-1));
+    arma::mat B0 = arma::randn<arma::mat>(J,K);
+    arma::mat B1 = 2.*arma::randu<arma::mat>(J,K);
+    arma::mat MU = X*Gamma;
+    arma::mat Qstar(J,K);
+    //double v0(.5),v1(2.);//,omega(.5);
+    double a0(1.),b0(1.),a1(1.),b1(1.);
+
+    arma::vec d0 = arma::ones<arma::vec>(nClass);
+    arma::vec pis = rDirichlet(d0);
+    arma::mat rstar=arma::ones<arma::mat>(J,K);
+    arma::vec pistar=arma::randu<arma::vec>(J);
+    arma::mat X_biject=arma::ones<arma::mat>(N,J);
+
+    //Start Markov chain
+    for(unsigned int t = 0; t < chain_length; t++){
+        parm_update_nomiss(N,J,K,nClass,Y,X_biject,B0,B1,CLASS,pis,Q,MU,vv,CLtable,v0,v1,d0,pistar,rstar,a0,b0,a1,b1,Qstar,cv1,cv0);
+        //Rcpp::List step_1 =parm_update_nomiss(N,J,K,nClass,Y,X_biject,B0,B1,CLASS,pis,Q,MU,vv,CLtable,v0,v1,d0,pistar,rstar,a0,b0,a1,b1,Qstar,cv1);
+        //v0 = Rcpp::as< double >(step_1[0]);
+        //v1 = Rcpp::as< double >(step_1[1]);
+        //omega = Rcpp::as< double >(step_1[2]);
+
+        arma::mat MU = X*Gamma;
+
+        //Update Gamma & deltas
+        Rcpp::List step_MVN = update_Gamma_Delta_MVN(N,V,K,J,Qstar,Gamma,deltas,X,diagXpX,w2,nu,bnu);
+        w2 = Rcpp::as< double >(step_MVN[0]);
+        nu = Rcpp::as< double >(step_MVN[1]);
+
+        if(t>burnin-1){
+            tmburn = t-burnin;
+            PISTAR.col(tmburn)  = pistar;
+            RSTAR.slice(tmburn) = rstar;
+            PIs.col(tmburn)      = pis;
+            QS.slice(tmburn)     = Q;
+            GAMMAS_OUT.slice(tmburn) = Gamma ;
+            Delta_tab+=deltas;
+            /*      B0s.slice(tmburn)     = B0;
```

Comment:
Need to likely obtain the OR_PPPs for model selection under the `auto_*()` case.